src/dex/jarvis-v6/jarvis-v6.ts

```diff
@@ -0,0 +1,473 @@
+      },
+    ];
+  }
+
+  // Encode params required by the exchange adapter
+  // Used for multiSwap, buy & megaSwap
+  // Hint: abiCoder.encodeParameter() could be useful
+  getAdapterParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: JarvisV6Data,
+    side: SwapSide,
+  ): AdapterExchangeParam {
+    const { swapFunction } = data;
+    const type = [JarvisSwapFunctions.MINT, JarvisSwapFunctions.REDEEM].indexOf(
+      swapFunction,
+    );
+
+    if (type === undefined) {
+      throw new Error(
+        `Jarvis: Invalid OpType ${swapFunction}, Should be one of ['mint', 'redeem']`,
+      );
+    }
+
+    const payload = this.abiCoder.encodeParameter(
+      {
+        ParentStruct: {
+          opType: 'uint',
+          expiration: 'uint128',
+        },
+      },
+      {
+        opType: type,
+        expiration: (Date.now() / 1000 + THIRTY_MINUTES).toFixed(0),
+      },
+    );
+
+    return {
+      targetExchange: data.poolAddress.toLowerCase(),
+      payload,
+      networkFee: '0',
+    };
+  }
+
+  // Encode call data used by simpleSwap like routers
+  // Used for simpleSwap & simpleBuy
+  // Hint: this.buildSimpleParamWithoutWETHConversion
+  // could be useful
+  async getSimpleParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: JarvisV6Data,
+    side: SwapSide,
+  ): Promise<SimpleExchangeParam> {
+    const { swapFunction } = data;
+    const timestamp = (Date.now() / 1000 + THIRTY_MINUTES).toFixed(0);
+
+    let swapFunctionParams: JarvisV6Params;
+    switch (swapFunction) {
+      case JarvisSwapFunctions.MINT:
+        swapFunctionParams = [
+          destAmount,
+          srcAmount,
+          timestamp,
+          this.augustusAddress,
+        ];
+        break;
+      case JarvisSwapFunctions.REDEEM:
+        swapFunctionParams = [
+          srcAmount,
+          destAmount,
+          timestamp,
+          this.augustusAddress,
+        ];
+        break;
+      default:
+        throw new Error(`Unknown function ${swapFunction}`);
+    }
+    const swapData = this.poolInterface.encodeFunctionData(swapFunction, [
+      swapFunctionParams,
+    ]);
+
+    return this.buildSimpleParamWithoutWETHConversion(
+      srcToken,
+      srcAmount,
+      destToken,
+      destAmount,
+      swapData,
+      data.poolAddress.toLowerCase(),
+    );
+  }
+
+  // Returns list of top pools based on liquidity. Max
+  // limit number pools should be returned.
+  async getTopPoolsForToken(
+    tokenAddress: Address,
+    limit: number,
+  ): Promise<PoolLiquidity[]> {
+    return [];
+  }
+
+  async getMaxTokensCapacity(
+    poolAddress: Address,
+    blockNumber: number,
+  ): Promise<bigint> {
+    const cacheKey = `${this.dexKey}_maxTokensCapacity_${poolAddress}`;
+    const cachedMaxTokensCapacity = await this.dexHelper.cache.get(
+      this.dexKey,
+      this.network,
+      cacheKey,
+    );
```

Comment:
I think the value should be cached in local memory as well for the same TTL at the time of request, otherwise when the remote cache expires, all instances may end up trying to repopulate it at the same time.

Reply:
I feel like this should be handled at upper level and abstracted away on calling `cache.(get|.setex)`